src/Files.App/Filesystem/RecentItems.cs

```diff
@@ -204,6 +205,46 @@ public Task<bool> UnpinFromRecentFiles(RecentItem item)
 			}));
 		}
 
+		public bool CheckIsRecentFilesEnabled()
+		{
+			using var subkey = Registry.CurrentUser.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer");
+			if (subkey is not null)
+			{
+				// quick access: show recent files option
+				bool showRecentValue = Convert.ToBoolean(subkey.GetValue("ShowRecent", true)); // 1 by default
+				if (!showRecentValue)
+				{
+					return false;
+				}
+			}
+
+			using var advSubkey = Registry.CurrentUser.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced");
+			if (advSubkey is not null)
+			{
+				// settings: personalization > start > show recently opened items
+				bool startTrackDocsValue = Convert.ToBoolean(advSubkey.GetValue("Start_TrackDocs", true)); // 1 by default
+				if (!startTrackDocsValue)
+				{
+					return false;
+				}
+			}
+
+			// for users in group policies
+			using var userPolicySubkey = Registry.CurrentUser.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\Explorer");
+			using var sysPolicySubkey = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\Explorer");
```

Comment:
Might be good to move to the top of the method. Same for the other usings.

Reply:
done! considering most people have recent items on, the happy path goes through all conditionals anyway